src/EPR.RegulatorService.Facade.Core/Clients/AntivirusClient.cs

```diff
@@ -0,0 +1,43 @@
+﻿using EPR.RegulatorService.Facade.Core.Models.TradeAntiVirus;
+using Microsoft.Extensions.Logging;
+using System.Text.Json;
+using System.Text;
+
+namespace EPR.RegulatorService.Facade.Core.Clients
+{
+    public class AntivirusClient : IAntivirusClient
+    {
+        private readonly HttpClient _httpClient;
+        private readonly ILogger<AntivirusClient> _logger;
+
+        public AntivirusClient(HttpClient httpClient, ILogger<AntivirusClient> logger)
+        {
+            _httpClient = httpClient;
+            _logger = logger;
+        }
+
+        public async Task<HttpResponseMessage> SendFile(FileDetails fileDetails, string fileName, MemoryStream fileStream)
+        {
+            try
+            {
+                fileDetails.Content = Convert.ToBase64String(fileStream.ToArray());
+
+                var jsonRequest = JsonSerializer.Serialize(fileDetails);
+                var stringContent = new StringContent(jsonRequest, Encoding.UTF8, "application/json");
+
+                var response = await _httpClient.PutAsync($"SyncAV/{fileDetails.Collection}/{fileDetails.Key}", stringContent);
+
+                response.EnsureSuccessStatusCode();
+
+                _logger.LogInformation("The file was successfully sent to antivirus app");
+
+                return response;
```

Comment:
What is the response here? If the response contains a result like 'clean' or 'infected' it would be better to rename this method to something like "VirusScanFile". If the file is actually being sent (and stays on the server) then perhaps SendFile is OK.

Reply:
on dead level only sent file status evaluate on controller level

Reply:
Given that it is scanning the file e.g. `if (antiVirusResult == ContentScan.Clean)` I would say rename this method to ScanFile